src/oop/static_modifier/medium/q4/MainTest.java

```diff
@@ -0,0 +1,119 @@
+package oop.static_modifier.medium.q4;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertArrayEquals;
+import static global.tools.CustomAssertions._assertEquals;
+import static global.utils.ArrayUtil.generateRandomArray;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    ObjectTest testClass;
+    String className = "ChemistryLab";
+    String firstDoubleName = "liquid1";
+    String secondDoubleName = "liquid2";
+    String firstIntName = "MIX_1";
+    String secondIntName = "MIX_2";
+    String arrayName = "guide";
+    String methodName = "mixFun";
+    static double initialValue1 = 10.5;
+    static double initialValue2 = 10.5;
+    int typeValue1 = 0; // Must be less than 5
+    int typeValue2 = 1; // Must be less than 5
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String circleString = "oop.static_modifier.medium.q4." + className;
+        testClass = new ObjectTest(circleString);
+        String modificationErrorMessage = "You have modified the provided portions of " + className + " class for the %s attribute. Please revert them to the original state.";
+        assertTrue(testClass.hasField(firstDoubleName, double.class, new String[]{"private", "static"}),
+                String.format(modificationErrorMessage,firstDoubleName));
+        assertTrue(testClass.hasField(secondDoubleName, double.class, new String[]{"private", "static"}),
+                String.format(modificationErrorMessage, secondDoubleName));
+        assertTrue(testClass.hasField(firstIntName, int.class, new String[]{"private", "final", "static"}),
+                String.format(modificationErrorMessage, firstIntName));
+        assertTrue(testClass.hasField(secondIntName, int.class, new String[]{"private", "final", "static"}),
+                String.format(modificationErrorMessage, secondIntName));
+        assertTrue(testClass.hasField(arrayName, double[].class, new String[]{"private"}),
+                String.format(modificationErrorMessage, arrayName));
+
+        for (int i = 0; i < 10; i++) {
+            double[] test = generateRandomArray(0.5, 20.0, ((int) (Math.random() * 20)));
+            Object[][] arguments = {{test, double[].class}};
+            Object classInstance = testClass.createInstance(arguments);
+            _assertArrayEquals(test, testClass.getFieldValue(classInstance, arrayName),
+                    String.format(modificationErrorMessage, className));
```

Comment:
This appears to be testing the constructor, so it should get a different error message since the current one specifies attribute